### PR TITLE
Add empty previous block id for fork status check - Closes #5486

### LINKED ISSUE #5486: Synchronization should end successfully
### Expected behavior
Block synchronization should end successfully

### Actual behavior
```
Message: The "otherBuffer" argument must be an instance of Buffer or Uint8Array. Received undefined
 Trace: TypeError [ERR_INVALID_ARG_TYPE]: The "otherBuffer" argument must be an instance of Buffer or Uint8Array. Received undefined
    at Uint8Array.equals (buffer.js:809:11)
    at Object.exports.isValidBlock (/lisk-sdk/elements/lisk-bft/src/fork_choice_rule.ts:46:15)
    at BFT.forkChoice (/lisk-sdk/elements/lisk-bft/src/bft.ts:136:22)
    at /lisk-sdk/framework/src/application/node/block_processor_v2.ts:140:20
    at Pipeline.run (/lisk-sdk/framework/src/application/node/processor/pipeline.ts:64:25)
    at Processor.forkStatus (/lisk-sdk/framework/src/application/node/processor/processor.ts:251:36)
    at BlockSynchronizationMechanism._computeBestPeer (/lisk-sdk/framework/src/application/node/synchronizer/block_synchronization_mechanism.ts:579:49)
    at BlockSynchronizationMechanism.run (/lisk-sdk/framework/src/application/node/synchronizer/block_synchronization_mechanism.ts:86:32)
    at Synchronizer.run (/lisk-sdk/framework/src/application/node/synchronizer/synchronizer.ts:138:25)
    at runMicrotasks (<anonymous>)
```

### Steps to reproduce
- One node forging all blocks
- Start another node from scratch wait for sync. 
- Stop the syncing node and wait for a while and start the node again

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #5486 

### How was it solved?

- Original implementation expected to be not equal with string comparison, but now since it's buffer, it should be buffer.

### How was it tested?

- Link with Lisk-core
- Make a node one block behind from another then start syncing again
- it should not fail syncing

### CODE REVIEW

File: framework/src/application/node/transport/schemas.ts [hunk 1]
```diff
@@ -34,7 +34,6 @@ export const schemas = {
 					type: 'string',
 					format: 'base64',
 				},
-				minItems: 1,
```
Comment (pablitovicente) [inline framework/src/application/node/transport/schemas.ts:37]: shouldn't this be kept if it's expecting an array of transactionIds?

Comment (shuse2) [reply framework/src/application/node/transport/schemas.ts]: ah sorry this was part of #5484 (wrong target branch)

File: framework/src/application/node/synchronizer/block_synchronization_mechanism.ts [hunk 2]
```diff
@@ -571,6 +571,7 @@ export class BlockSynchronizationMechanism extends BaseSynchronizer {
 			id: Buffer.alloc(0),
 			height: selectedPeers[randomPeerIndex].height,
 			version: selectedPeers[randomPeerIndex].blockVersion,
+			previousBlockID: Buffer.alloc(0),
```
Comment (ManuGowda) [inline framework/src/application/node/synchronizer/block_synchronization_mechanism.ts:574]: if we always send empty `previousBlockID`. why not ignore this if its not required?

Comment (shuse2) [reply framework/src/application/node/synchronizer/block_synchronization_mechanism.ts:574]: It is comparing previousBlockID in the fork choice rule to decide wether to sync or not.
Previously it was string, so comparing with `undefined` was fine, but now it should be at least Buffer to use `buf.equal`

**Pull request merged**